src/main/java/gregicadditions/machines/TileEntityProcessingArray.java

```diff
@@ -66,123 +84,101 @@ public MetaTileEntity createMetaTileEntity(MetaTileEntityHolder holder) {
-				case "electric_furnace":
-					return RecipeMaps.FURNACE_RECIPES;
-				case "bender":
-					return RecipeMaps.BENDER_RECIPES;
-				case "arc_furnace":
-					return RecipeMaps.ARC_FURNACE_RECIPES;
-				case "autoclave":
-					return RecipeMaps.AUTOCLAVE_RECIPES;
-				case "assembler":
-					return RecipeMaps.ASSEMBLER_RECIPES;
-				case "brewery":
-					return RecipeMaps.BREWING_RECIPES;
-				case "canner":
-					return RecipeMaps.CANNER_RECIPES;
-				case "chemical_bath":
-					return RecipeMaps.CHEMICAL_BATH_RECIPES;
-				case "chemical_reactor":
-					return RecipeMaps.CHEMICAL_RECIPES;
-				case "compressor":
-					return RecipeMaps.COMPRESSOR_RECIPES;
-				case "cutter":
-					return RecipeMaps.CUTTER_RECIPES;
-				case "distillery":
-					return RecipeMaps.DISTILLATION_RECIPES;
-				case "electromagnetic_separator":
-					return RecipeMaps.ELECTROMAGNETIC_SEPARATOR_RECIPES;
-				case "fermenter":
-					return RecipeMaps.FERMENTING_RECIPES;
-				case "fluid_canner":
-					return RecipeMaps.FLUID_CANNER_RECIPES;
-				case "fluid_heater":
-					return RecipeMaps.FLUID_HEATER_RECIPES;
-				case "fluid_solidifier":
-					return RecipeMaps.FLUID_SOLIDFICATION_RECIPES;
-				case "forge_hammer":
-					return RecipeMaps.FORGE_HAMMER_RECIPES;
-				case "forming_press":
-					return RecipeMaps.FORMING_PRESS_RECIPES;
-				case "microwave":
-					return RecipeMaps.MICROWAVE_RECIPES;
-				case "mixer":
-					return RecipeMaps.MIXER_RECIPES;
-				case "packer":
-					return RecipeMaps.PACKER_RECIPES;
-				case "unpacker":
-					return RecipeMaps.UNPACKER_RECIPES;
-				case "plasma_arc_furnace":
-					return RecipeMaps.PLASMA_ARC_FURNACE_RECIPES;
-				case "polarizer":
-					return RecipeMaps.POLARIZER_RECIPES;
-				case "laser_engraver":
-					return RecipeMaps.LASER_ENGRAVER_RECIPES;
-				case "wiremill":
-					return RecipeMaps.WIREMILL_RECIPES;
-				case "mass_fab":
-					return GARecipeMaps.MASS_FAB_RECIPES;
-				case "replicator":
-					return GARecipeMaps.REPLICATOR_RECIPES;
-				case "sifter":
-					return RecipeMaps.SIFTER_RECIPES;
-				case "extruder":
-					return RecipeMaps.EXTRUDER_RECIPES;
-				case "bundler":
-					return GARecipeMaps.BUNDLER_RECIPES;
-				default:
-					return null;
+		protected Recipe findRecipe(long maxVoltage, IItemHandlerModifiable inputs, IMultipleTankHandler fluidInputs, Tuple<ItemStack, RecipeMap<?>> itemStackRecipeMapTuple) {
+			if(itemStackRecipeMapTuple == null) {
+				return null;
 			}
 
+			this.machineItemStack = itemStackRecipeMapTuple.getFirst();
+			this.recipeMap = itemStackRecipeMapTuple.getSecond();
+
+			return findRecipe(maxVoltage, inputs, fluidInputs);
 		}
 
 		@Override
 		protected Recipe findRecipe(long maxVoltage,
 									IItemHandlerModifiable inputs,
 									IMultipleTankHandler fluidInputs) {
 
-			String machineName = findMachine(inputs);
-			RecipeMap<?> recipeMap = getRecipeMaps(machineName);
 
-			// No valid recipe map.
-			if(recipeMap == null)
```

Comment:
does this work? it seems like `findRecipe` would now behave incorrectly if the other override isn't called first

Reply:
`findRecipe` is always called from within `trySearchNewRecipe`, which means that the recipe map, if is exists, would always be added to the Tuple, so it should be fine.

Reply:
It's probably true at the moment, but I'd be careful about this assumption. Someone could call it from elsewhere.